### PR TITLE
[heartbeat][libbeat][metricbeat][filebeat] Pass TLS options to forward proxies

### PR DESCRIPTION
**Pull request by andrewvc**
Fixes a bug where TLS options would not be passed to forward proxies. This fixes a reported issue where x509 client cert auth doesn't work due to the cert parameters not being passed in.

Since this doesn't give us a full dialer we can't easily record TLS certificate expiry with proxies with this approach, so those fields would be missing. I think it makes sense to merge this change as-is and  fix that secondary bug in a subsequent PR. I've opened a separate issue ( https://github.com/elastic/beats/issues/15797 ) to cover that.

I've tested this manually against with the following config: https://github.com/elastic/uptime-contrib/tree/master/testing/configs/client-auth , note that you'll need the included client certs in that directory, and will need to alter the YAML to point at their actual path.

I think in this scenario I'm fine with only having unit tests because the complexity of setting of a forward proxy + a custom TLS backend is very high.

Fixes https://github.com/elastic/beats/issues/15524

### CODE REVIEW

File: heartbeat/monitors/active/http/http.go [hunk 1]
```diff
@@ -138,6 +138,7 @@ func newRoundTripper(config *Config, tls *transport.TLSConfig) (*http.Transport,
 		Proxy:             proxy,
 		Dial:              dialer.Dial,
 		DialTLS:           tlsDialer.Dial,
+		TLSClientConfig:   tls.ToConfig(),
```
Comment (urso) [inline heartbeat/monitors/active/http/http.go]: Can you add this change to some other packages, please?

- `libbeat/output/elasticsearch`
- `metricbeat/helper`

Comment (andrewvc) [reply heartbeat/monitors/active/http/http.go]: @urso I've done so 4 additional spots in 5a4e09c

**Comment by andrewvc** [comment]
@urso should be ready for re-review, I added this setting to every other spot I saw `DialTLS` in the code.

File: heartbeat/monitors/active/http/http.go (hunk 1 above)
Comment (urso) [reply heartbeat/monitors/active/http/http.go]: Thank you.

**Pull request merged**